core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/OfflineFirstNewsRepository.kt

```diff
@@ -72,15 +74,25 @@ class OfflineFirstNewsRepository @Inject constructor(
             },
             modelDeleter = newsResourceDao::deleteNewsResources,
             modelUpdater = { changedIds ->
+                val userData = niaPreferencesDataSource.userData.first()
+                val hasOnBoarded = userData.shouldHideOnboarding
+                val followedTopicIds = userData.followedTopics
+
                 // TODO: Make this more efficient, there is no need to retrieve populated
                 //  news resources when all that's needed are the ids
-                val existingNewsResourceIds = newsResourceDao.getNewsResources(
-                    useFilterNewsIds = true,
-                    filterNewsIds = changedIds.toSet(),
-                )
-                    .first()
-                    .map { it.entity.id }
-                    .toSet()
+                val existingFollowedChangedNewsResourceIds = when {
+                    hasOnBoarded -> newsResourceDao.getNewsResources(
+                        useFilterTopicIds = true,
+                        filterTopicIds = followedTopicIds,
+                        useFilterNewsIds = true,
+                        filterNewsIds = changedIds.toSet(),
+                    )
+                        .first()
+                        .map { it.entity.id }
+                        .toSet()
+                    // No need to retrieve anything if notifications won't be sent
+                    else -> emptySet()
+                }
 
                 changedIds.chunked(SYNC_BATCH_SIZE).forEach { chunkedIds ->
```

Comment:
Not related to the current PR but a comment here would be useful: 

Obtain the news resources which have changed from the network and upsert them locally